spec/tbd_tests_spec.rb

```diff
@@ -4869,7 +5154,267 @@
+
+    # Fallback to ERROR # 2: no KHI for "Office Front Wall".
+    expect(io.has_key?(:khis)).to be(true)
+    expect(io[:khis].size).to eq(1)
+    expect(surfaces["Office Front Wall"].has_key?(:khis)).to be(false)
+
+    # ... concerning the "Office Left Wall" (underatable material).
+    left_office_wall = os_model.getSurfaceByName("Office Left Wall")
+    expect(left_office_wall.empty?).to be(false)
+    left_office_wall = left_office_wall.get
+    c = left_office_wall.construction.get.to_Construction.get
+    expect(c.numLayers).to eq(1)
+    #layer = c.getLayer(0).to_MasslessOpaqueMaterial
+    layer = c.getLayer(0).to_StandardOpaqueMaterial
+    expect(layer.empty?).to be(false)
+    layer = layer.get
+    expect(layer.name.get).to eq("'Office Left Wall' m tbd")
+    #expect(layer.thermalResistance).to be_within(0.001).of(0.001)
+    expect(layer.thermalConductivity).to be_within(0.1).of(3.0)
+    expect(layer.thickness).to be_within(0.001).of(0.003)
+    # Regardless of the targetted material type ('standard' vs 'massless'), TBD
+    # will ensure a minimal RSi value of 0.001 m2.K/W, i.e. no derating despite
+    # the surface having thermal bridges.
+    expect(surfaces["Office Left Wall"].has_key?(:heatloss)).to be(true)
+    expect(surfaces["Office Left Wall"][:heatloss]).to be_within(0.1).of(180)
+    expect(surfaces["Office Left Wall"].has_key?(:r_heatloss)).to be(true)
+    expect(surfaces["Office Left Wall"][:r_heatloss]).to be_within(0.1).of(180)
+
+    expect(surfaces["Fine Storage Right Wall"].has_key?(:heatloss)).to be(true)
+    expect(surfaces["Fine Storage Right Wall"].has_key?(:r_heatloss)).to be(false)
+    # ... concerning the new material (with a name already including " tbd").
+    # TBD ignores all such materials (a safeguard against iterative TBD
+    # runs). Contrary to the previous critical cases of highly conductive
+    # materials, TBD doesn't even try to set the :r_heatloss hash value - tough!
+    right_fine_wall = os_model.getSurfaceByName("Fine Storage Right Wall")
+    expect(right_fine_wall.empty?).to be(false)
+    right_fine_wall = right_fine_wall.get
+    c = right_fine_wall.construction.get.to_Construction.get
+    layer = c.getLayer(0).to_MasslessOpaqueMaterial
+    expect(layer.empty?).to be(false)
+    layer = layer.get
+    expect(layer.name.get).to eq("poor material m tbd")
+    expect(layer.thermalResistance).to be_within(0.1).of(1.0)
+
+    # Mimics TBD 'measure.rb' method 'exitTBD()'
+    unless TBD.logs.empty? || TBD.status < TBD.log_level
+      msgs = []
+      TBD.logs.each do |l|
+        msgs << "#{l[:time]} (#{TBD.tag(l[:level])}) #{l[:msg]}"
+      end
+      io[:logs] = msgs
+    end
+
+    out = JSON.pretty_generate(io)
+    outP = File.dirname(__FILE__) + "/../json/tbd_warehouse9.out.json"
+    File.open(outP, "w") do |outP|
+      outP.puts out
+    end
+    # ... should contain a 'logs' entry at the end of the JSON output file.
```

Comment:
For me this output just seems to show the log level, no message is included?

```
    "logs": [
        " (ERROR) ",
        " (ERROR) ",
        " (ERROR) ",
        " (ERROR) ",
        " (WARNING) ",
        " (WARNING) "
    ]
```

Reply:
Nice catch. A remnant. Fixed to better reflect what users would see (e.g., runner output, `tbd.out.json`).